packages/cli/src/build.ts

```diff
@@ -106,6 +106,45 @@ export function getFiles(
   return tsConfigFiles;
 }
 
+type GetInitialCompilerHostOptions = {
+  format: BuildType[];
+  compilerOptions: CompilerOptions;
+  system: System;
+  host?: CompilerHost;
+  projectReferences?: readonly ProjectReference[];
+};
+
+/**
+ * Get the initial compiler host to use for the build. This function will return
+ * the host to use based on the build options.
+ *
+ * @param options - The options.
+ * @param options.format - The formats to build.
+ * @param options.compilerOptions - The compiler options to use.
+ * @param options.system - The file system to use.
+ * @param options.projectReferences - The project references to use.
+ * @returns The initial compiler host to use for the build.
+ */
+function getInitialCompilerHost({
+  format,
+  compilerOptions,
+  system,
+  projectReferences,
+}: GetInitialCompilerHostOptions) {
+  const mockProgram = createProgram({
+    rootNames: [],
+    options: {},
+    projectReferences,
+  });
+
+  return createProjectReferencesCompilerHost(
```

Comment:
What about builds that don't use project references? What host did they use before?

Reply:
The default one, e.g., the result of `typescript.createCompilerHost`. They don't need a custom host.